### PR TITLE
Prefer matching widget file name

### PR DESCRIPTION
**Pull request by Konoshenko**
### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[ ] Bug fix
[X] New rule
[ ] Changes an existing rule
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,10 +14,10 @@
 [Anti-patterns](#anti-patterns)
 
 <img
-  src="https://raw.githubusercontent.com/dart-code-checker/dart-code-metrics/master/doc/.assets/logo.svg"
-  alt="Dart Code Metrics logo"
-  height="120" width="120"
-  align="right">
+src="https://raw.githubusercontent.com/dart-code-checker/dart-code-metrics/master/doc/.assets/logo.svg"
```
Comment (dkrutskikh) [inline README.md:17]: @Konoshenko please revert all changes in this file

File: test/analyzers/lint_analyzer/rules/rules_list/prefer_match_file_name/examples/example.dart [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class Example extends StatelessWidget {
```
Comment (dkrutskikh) [inline test/analyzers/lint_analyzer/rules/rules_list/prefer_match_file_name/examples/example.dart:1]: @Konoshenko please add examples with statefullwidget

File: lib/src/analyzers/lint_analyzer/rules/rules_list/prefer_match_file_name/prefer_match_file_name.dart [hunk 3]
```diff
@@ -0,0 +1,46 @@
+import 'package:analyzer/dart/ast/ast.dart';
```
Comment (dkrutskikh) [inline lib/src/analyzers/lint_analyzer/rules/rules_list/prefer_match_file_name/prefer_match_file_name.dart:1]: @Konoshenko please add your rule in rules_factory

File: lib/src/analyzers/lint_analyzer/rules/rules_list/prefer_match_file_name/visitor.dart [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+  String _formatClassName(String className) {
+    final exp = RegExp('(?<=[a-z])[A-Z]');
+    final result =
+        className.replaceAllMapped(exp, (m) => '_${m.group(0)!}').toLowerCase();
+
+    return result;
+  }
+
+  String getFileName(String filePath) =>
```
Comment (dkrutskikh) [inline lib/src/analyzers/lint_analyzer/rules/rules_list/prefer_match_file_name/visitor.dart:31]: @Konoshenko please use `path` package

**Review by dkrutskikh — CHANGES_REQUESTED** [review]

**Comment by dkrutskikh** [comment]
@Konoshenko first of all - thank you for contribute. Please take a look at review notes.

File: lib/src/analyzers/lint_analyzer/rules/rules_factory.dart [hunk 5]
```diff
@@ -1,3 +1,5 @@
+import 'package:dart_code_metrics/src/analyzers/lint_analyzer/rules/rules_list/prefer_match_file_name/prefer_match_file_name.dart';
```
Comment (dkrutskikh) [inline lib/src/analyzers/lint_analyzer/rules/rules_factory.dart:1]: @Konoshenko please use relative import

**Review by dkrutskikh — CHANGES_REQUESTED** [review]

**Review by dkrutskikh — APPROVED** [review]

**Review by incendial — CHANGES_REQUESTED** [review]
Thank you for a great work! Left some small comments.

**Review by dkrutskikh — APPROVED** [review]

**Review by incendial — APPROVED** [review]

**Pull request merged**